src/core/client/auth/views/SignUp/SignUpWithEmail.tsx

```diff
@@ -1,68 +1,76 @@
-import { Localized } from "@fluent/react/compat";
+import { Form, Formik } from "formik";
 import React, { FunctionComponent } from "react";
-import { Form } from "react-final-form";
+import * as Yup from "yup";
 
-import EmailField from "coral-auth/components/EmailField";
-import SetPasswordField from "coral-auth/components/SetPasswordField";
-import UsernameField from "coral-auth/components/UsernameField";
-import { FormError, OnSubmit } from "coral-framework/lib/form";
-import { Flex, Icon } from "coral-ui/components/v2";
-import { Button, CallOut } from "coral-ui/components/v3";
+import {
+  PASSWORD_MIN_LENGTH,
+  USERNAME_MAX_LENGTH,
+  USERNAME_MIN_LENGTH,
+  USERNAME_REGEX,
+} from "coral-common/helpers/validate";
+import {
+  INVALID_CHARACTERS,
+  PASSWORD_TOO_SHORT,
+  USERNAME_TOO_LONG,
+  USERNAME_TOO_SHORT,
+  VALIDATION_REQUIRED,
+} from "coral-framework/lib/messages";
+
+import EmailField from "./EmailField";
+import SetPasswordField from "./SetPasswordField";
+import SignupError from "./SignupError";
+import SignUpSubmit from "./SignUpSubmit";
+import UsernameField from "./UsernameField";
 
 import styles from "./SignUpWithEmail.css";
 
-interface FormProps {
+interface FormValues {
   email: string;
   username: string;
   password: string;
-  confirmPassword: string;
 }
 
-interface FormSubmitProps extends FormProps, FormError {}
-
 interface Props {
-  onSubmit: OnSubmit<FormSubmitProps>;
+  onSubmit: (input: any) => void;
 }
 
+const SignupValidationSchema = Yup.object().shape({
```

Comment:
`Yup` is a fork of `Joi` that is more client-side friendly. We can move form validation from each field to a single validation schema if wanted, the schema can be passed through to the `Formik` component and will validate touched fields on blur and submit.